depends/toolchain.cmake.in

```diff
@@ -117,11 +117,11 @@ unset(pkg_config_path)
 set(PKG_CONFIG_ARGN --static)
 
 
-# Ensure that the docstrings in the following `set(... CACHE ...)`
-# commands match those in the root CMakeLists.txt file.
-
-if(NOT WITH_GUI AND "@no_qt@" STREQUAL "1")
-  set(WITH_GUI OFF CACHE STRING "Build GUI.")
+# Set configuration options for the main build system.
+if("@no_qt@")
```

Comment:
I wonder if we should depart from upstream/master behavior for all of these options to be more CMake-like?

(I'm not just referring to qt here, but all depends options)

What if depends doesn't turn anything on by default? Meaning.. quit forwarding any meaning from built packages into Core's build. So, you can build depends with any set of packages/options, but that doesn't turn anything just on by using the generated toolchain file.

That way, if I build depends with qt, I still have to specify `WITH_GUI` in order to get bitcoin-qt. Conversely, if I build depends with `NO_QT` and build with `WITH_GUI`, that would be an error.

Thoughts? @hebasto @fanquake @laanwj

Reply:
> ... to be more CMake-like?

In the future, CMake can manage all dependencies using its own modules like `ExternalProject` or `FetchContent`.